duckduckgo/iOS
# Add support for syncing Credentials

Task/Issue URL: https://app.asana.com/0/72649045549333/1204029872658872/f
Tech Design URL: https://app.asana.com/0/481882893211075/1204906306298609/f

**Description**:
Update Autofill UI data models and related code to account's domain and username being optional.
Move Bookmarks Database Cleaner to SyncBookmarksAdapter.
Add SyncCredentialsAdapter (and it also contains CredentialsDatabaseCleaner).
Added sync triggers to Autofill UI and fixed UI for situations when user is editing an Autofill Login
and sync causes Autofill data to refresh.



**Steps to test this PR**:
1. Run release app, enable sync and have your bookmarks synced.
2. Populate some autofill logins.
3. Run the app from Xcode on top of the release app.
4. Verify that your logins are synced.
5. Play with Autofill and verify that items are synced according to the specification.



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
